core/domain/stats_jobs_one_off.py

```diff
@@ -127,3 +143,2369 @@ def reduce(key, stringified_values):
+        #   {{ choices[answer|int] }}
+        interaction = state.interaction
+        if rule_str == cls._DEFAULT_RULESPEC_STR or (
+                rule_spec.rule_type == 'Equals'):
+            customization_args = interaction.customization_args
+            choices = customization_args['choices']['value']
+
+            # An answer group index is only available for non-default rules.
+            if rule_str != cls._DEFAULT_RULESPEC_STR:
+                outcome = interaction.answer_groups[answer_group_index].outcome
+                temporary_special_param = next(
+                    (param_change for param_change in outcome.param_changes
+                    if param_change.name == (
+                        AnswerMigrationJob._TEMPORARY_SPECIAL_RULE_PARAMETER)),
+                    None)
+
+            # If the subject type is not an answer, then the index is extracted
+            # by matching the answer_str against possible choices in the
+            # interaction customization args. The index was not stored in the
+            # rule spec string. Otherwise, if this is a default classified
+            # answer the only way to reconstitute the index is using the
+            # submitted answer itself.
+            if rule_str == cls._DEFAULT_RULESPEC_STR or temporary_special_param:
+                if answer_str not in choices:
+                    return (
+                        None,
+                        'Answer \'%s\' was not found among the choices in the '
+                        'exploration: %s' % (answer_str, choices))
+                clicked_index = choices.index(answer_str)
+            else:
+                # Extract the clicked index from the rule string.
+                clicked_index = int(rule_str[len(rule_spec.rule_type) + 1:-1])
+                if clicked_index >= len(choices):
+                    return (
+                        None,
+                        'Clicked index %d is out of bounds for corresponding '
+                        'choices in the exploration: %s (len=%d)' % (
+                            clicked_index, choices, len(choices)))
+                if answer_str != choices[clicked_index]:
+                    rte_latex_start= (
+                        '<oppia-noninteractive-math '
+                        'raw_latex-with-value="&amp;quot;')
+                    rte_latex_end = '&amp;quot;"></oppia-noninteractive-math>'
+                    expected_answer = choices[clicked_index]
+                    if (not answer_str.startswith('$') or
+                            not answer_str.endswith('$') or
+                            not expected_answer.startswith(rte_latex_start) or
+                            not expected_answer.endswith(rte_latex_end)):
+                        stripped_answer_str = answer_str[1:-1]
+                        stripped_expected_answer = expected_answer[
+                            len(rte_latex_s
```

Comment:
Yes. :( It's because of the allocation-happy StateRuleAnswerLogModel that will create new models for default and fuzzy rules just because we check for them, not because they're storing any values. Fuzzy rules are actually stripped out in the purge job, but this is kept around as a guard. I might remove it.